add nats health check (#891)
Xabaril/AspNetCore.Diagnostics.HealthChecks: src/HealthChecks.Nats/README.md

# NATS Health Check

This health check verifies the ability to communicate with a [NATS server](https://nats.io/about/). \
It relies on `NATS.Client` package version `0.14.3` at the time this was developed. \
Latest tag for the [official dockerhub image](https://hub.docker.com/_/nats/) is `2.6.6`.

## Builder Extension

```cs
public void ConfigureServices(IServiceCollection services)
{
    services
        .AddHealthChecks()

>         .AddNats((NatsOptions setup) => {
>             setup.Url = "A string containing the URL (or URLs) to the NATS Server.";

Reviewer: ```suggestion
        .AddNats((NatsOptions setup) =>
        {
            setup.Url = "A string containing the URL (or URLs) to the NATS Server.";
```

>         .AddNats((NatsOptions setup) => {
>             setup.Url = "A string containing the URL (or URLs) to the NATS Server.";
>             setup.CredentialsPath = "The full path to a chained credentials file.";
>             setup.Jwt = "The path to a user's public JWT credentials.";
>             setup.PrivateNKey = "The path to a file for user user's private Nkey seed.";
>         });

Reviewer: ```suggestion
        .AddNats(options =>
        {
            options.Url = "A string containing the URL (or URLs) to the NATS Server.";
            options.CredentialsPath = "The full path to a chained credentials file.";
            options.Jwt = "The path to a user's public JWT credentials.";
            options.PrivateNKey = "The path to a file for user user's private Nkey seed.";
        });
```

# NATS Health Check

This health check verifies the ability to communicate with a [NATS server](https://nats.io/about/). \

> It relies on `NATS.Client` package version `0.14.3` at the time this was developed. \

Reviewer: ```suggestion
It relies on `NATS.Client` package. \
```

Latest tag for the [official dockerhub image](https://hub.docker.com/_/nats/) is `2.6.6`.

## Builder Extension

> ```cs

Reviewer: ```suggestion
```csharp
```

public void ConfigureServices(IServiceCollection services)
{
    services
        .AddHealthChecks()
        .AddNats(options =>
        {

>             options.Url = "A string containing the URL (or URLs) to the NATS Server.";

Reviewer: It would be better to provide a concrete example values (and move readable descriptions into comments).

Author: Commit: 403cd2af817975ded7ebe3415066f979af859ebb [403cd2a]
Parents: 7258b08ed0
nullable enable, readme
